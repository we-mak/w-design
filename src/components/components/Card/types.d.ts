import { GlobProps } from "../../../typings";

export interface CardProps extends GlobProps {
  children?: React.ReactChild | React.ReactChild[];
  raised?: boolean;
  isLoading?: boolean;
}

export interface CardMediaProps extends GlobProps {
  // image cover url
  imageUrl?: string;
  children?: React.ReactChild;
}

export interface CardHeaderProps extends GlobProps {
  title?: string;
  description?: string | React.ReactChild;
  children?: React.ReactChild;
}
