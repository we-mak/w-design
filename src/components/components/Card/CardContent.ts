import styled from "styled-components";

export const CardContent = styled.div`
  flex: 1 1 auto;
  padding: 0.8rem;
  &:not(:last-child) {
    padding-bottom: 0;
  }
`;
