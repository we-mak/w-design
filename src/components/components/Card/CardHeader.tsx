import * as React from "react";
import styled from "styled-components";
import Typo from "../../elements/Typo";
import { CardHeaderProps } from "./types";

const CardHeaderWrapper = styled.div`
  padding: 0.8rem;
  padding-bottom: 0;
  h3 {
    font-weight: 500;
  }
`;
CardHeaderWrapper.displayName = "CardHeaderComponent";

export const CardHeader = ({ title, ...rest }: CardHeaderProps) => {
  return (
    <CardHeader {...rest}>
      <Typo appearance="h4" tag="h3">
        {title}
      </Typo>
    </CardHeader>
  );
};
