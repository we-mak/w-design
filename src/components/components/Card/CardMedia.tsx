import * as React from "react";
import styled from "styled-components";
import { borderRadius } from "styled-system";
// import { CardMediaProps } from "./types";
import Picture from "../../elements/Picture";

const CardMediaWrapper = styled.div<any>`
  &:first-child {
    padding-top: 0;
    img {
      border-top-left-radius: ${props => props.theme.radii[2]};
      border-top-right-radius: ${props => props.theme.radii[2]};
      ${borderRadius}
    }
  }
`;
CardMediaWrapper.displayName = "CardMediaWrapper";

const CoverImage = styled(Picture as any)`
  width: 100%;
  max-width: 100%;
  height: auto;
  display: block;
`;
CoverImage.displayName = "CoverImage";

export const CardMedia = ({ srcSets, imageUrl, alt, children, ...rest }: any) => {
  return (
    <CardMediaWrapper {...rest}>
      {imageUrl && <CoverImage dataSrc={imageUrl} srcSets={srcSets} alt={alt} />}
      {children}
    </CardMediaWrapper>
  );
};
