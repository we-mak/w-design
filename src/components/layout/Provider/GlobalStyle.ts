import { createGlobalStyle } from "styled-components";
import theme from "../../../common/styleUtils/theme";
const colors = theme.colors;

/*! normalize.css v8.0.1 | MIT License | github.com/necolas/normalize.css */
export const GlobalStyle = createGlobalStyle`
html {
  line-height: 1.15;
  -webkit-text-size-adjust: 100%;
}
body {
  margin: 0;
}
main {
  display: block;
}
h1 {
  font-size: 2em;
  margin: 0.67em 0;
}
hr {
  box-sizing: content-box;
  height: 0;
  overflow: visible;
}
pre {
  font-family: monospace, monospace;
  font-size: 1em;
}
a {
  background-color: transparent;
}
abbr[title] {
  border-bottom: none;
  text-decoration: underline;
  text-decoration: underline dotted;
}
b,
strong {
  font-weight: bolder;
}
code,
kbd,
samp {
  font-family: monospace, monospace;
  font-size: 1em;
}
small {
  font-size: 80%;
}
sub,
sup {
  font-size: 75%;
  line-height: 0;
  position: relative;
  vertical-align: baseline;
}
sub {
  bottom: -0.25em;
}
sup {
  top: -0.5em;
}
img {
  border-style: none;
}
button,
input,
optgroup,
select,
textarea {
  font-family: inherit;
  font-size: 100%;
  line-height: 1.15;
  margin: 0;
}
button,
input {
  overflow: visible;
}
button,
select {
  text-transform: none;
}
[type="button"],
[type="reset"],
[type="submit"],
button {
  -webkit-appearance: button;
}
[type="button"]::-moz-focus-inner,
[type="reset"]::-moz-focus-inner,
[type="submit"]::-moz-focus-inner,
button::-moz-focus-inner {
  border-style: none;
  padding: 0;
}
[type="button"]:-moz-focusring,
[type="reset"]:-moz-focusring,
[type="submit"]:-moz-focusring,
button:-moz-focusring {
  outline: 1px dotted ButtonText;
}
fieldset {
  padding: 0.35em 0.75em 0.625em;
}
legend {
  box-sizing: border-box;
  color: inherit;
  display: table;
  max-width: 100%;
  padding: 0;
  white-space: normal;
}
progress {
  vertical-align: baseline;
}
textarea {
  overflow: auto;
}
[type="checkbox"],
[type="radio"] {
  box-sizing: border-box;
  padding: 0;
}
[type="number"]::-webkit-inner-spin-button,
[type="number"]::-webkit-outer-spin-button {
  height: auto;
}
[type="search"] {
  -webkit-appearance: textfield;
  outline-offset: -2px;
}
[type="search"]::-webkit-search-decoration {
  -webkit-appearance: none;
}
::-webkit-file-upload-button {
  -webkit-appearance: button;
  font: inherit;
}
details {
  display: block;
}
summary {
  display: list-item;
}
template {
  display: none;
}
[hidden] {
  display: none;
}

*,
*::before,
*::after {
  box-sizing: inherit;
}

html {
  box-sizing: border-box;
  font-size: 20px;
  line-height: 1.5;
  -webkit-tap-highlight-color: transparent;
}

body {
  font-size: .8rem;
  overflow-x: hidden;
  text-rendering: optimizeLegibility;
}


h1 {
    font-size: 1.6rem;
  }

  h2 {
    font-size: 1.4rem;
  }

  h3 {
    font-size: 1.2rem;
  }

  h4 {
    font-size: 1rem;
  }
  h5 {
    font-size: 0.9rem;
  }

  h6 {
    font-size: 0.8rem;
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-weight: 700;
    line-height: 1.25;
    margin-bottom: 0.5em;
    margin-top: 0;
    font-family: ${(props: any) => props.theme.fonts.medium};
  }

  h1,
  h2,
  h3,
  h4 {
    color: ${colors.N50};
  }

  h5,
  h6 {
    color: ${colors.N60};
  }

  p {
    color: ${colors.N70};
    line-height: 1.5;
    margin-bottom: 0.5em;
    margin-top: 0;
    font-family: ${(props: any) => props.theme.fonts.normal};
  }

  a {
    font-family: ${(props: any) => props.theme.fonts.normal};
    color: ${colors.B50};
    outline: none;
    text-decoration: none;

    &:hover {
      color: ${colors.B30};
    }

    &:focus,
    &:hover,
    &:active {
      text-decoration: underline;
    }
    &:focus,
    &:active {
      color: ${colors.B60};
    }
    &:focus {
      text-decoration: none;
    }
  }

  b,
  strong {
    font-weight: bolder;
  }

  small {
    font-size: 80%;
    font-weight: 400;
    color: ${colors.N30};
  }

  blockquote {
    margin-left: 0;
    padding: 0.4rem 0.8rem;
    color: ${colors.N30};
    p:last-child {
      margin-bottom: 0;
    }
  }

  table {
    border-collapse: collapse;
    border-spacing: 0;
    text-align: left;
    width: 100%;
  }
  th {
    border-bottom: 0.05rem solid ${props => props.theme.colors["N5"]};
    padding: 0.6rem 0.4rem;
    border-bottom-width: 0.1rem;
  }
  td {
    border-bottom: 0.05rem solid ${props => props.theme.colors["N2"]};
    padding: 0.6rem 0.4rem;
  }
`;
