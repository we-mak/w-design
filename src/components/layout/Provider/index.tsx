import * as React from "react";
import { ThemeProvider } from "styled-components";
import defaultTheme from "../../../common/theme";
import { ResetCss } from "./ResetCss";
import Root from "./Root";

export interface Props extends React.HTMLProps<HTMLDivElement> {
  theme?: any;
}

class Provider extends React.Component<Props> {
  stylesheet: any;

  static displayName = "WProvider";

  public static defaultProps: Props = {
    theme: {}
  };

  componentDidMount() {
    this.stylesheet = document.createElement("style");
    this.stylesheet.type = "text/css";
    // this.stylesheet.setAttribute("rel", "preload");
    // this.stylesheet.setAttribute("as", "style");
    this.stylesheet.innerHTML = ResetCss;

    if (document && document.head) {
      document.head.appendChild(this.stylesheet);
    }
  }

  componentWillUnmount() {
    if (this.stylesheet && document && document.head) {
      document.head.removeChild(this.stylesheet);
      delete this.stylesheet;
    }
  }

  render() {
    const { theme, ...props } = this.props;
    return (
      <ThemeProvider theme={{ ...defaultTheme, ...theme }}>
        <Root {...props as any} />
      </ThemeProvider>
    );
  }
}

export default Provider;
