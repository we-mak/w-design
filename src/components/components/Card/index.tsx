import * as React from "react";
import styled, { css } from "styled-components";
import { space, border, borderRadius } from "styled-system";
import { CardHeader } from "./CardHeader";
import { CardMedia } from "./CardMedia";
import { CardContent } from "./CardContent";
import { CardProps } from "./types";

const getCardStyle = (props: CardProps) => {
  return css`
    display: flex;
    flex-direction: column;
    background: #fff;
    border-radius: ${props.theme.radii[2]};
    box-shadow: ${props.theme.shadows[2]};

    ${props.raised &&
      `
    transition: box-shadow 0.2s cubic-bezier(0.5, 0.045, 0.4, 1);
    &:hover {
      box-shadow: ${props.theme.shadows[4]}
    }
    `}
  `;
};

const CardWrapper = styled.div`
  ${getCardStyle}
  ${space}
  ${border}
  ${borderRadius}
`;
CardWrapper.displayName = "CardWrapper";

const Card = ({ children, ...rest }: CardProps) => {
  return <CardWrapper {...rest}>{children}</CardWrapper>;
};

const CardFooter = styled.div`
  padding: 0.8rem;
`;
CardFooter.displayName = "CardFooter";

export default Card;
export { CardMedia, CardHeader, CardContent, CardFooter };
