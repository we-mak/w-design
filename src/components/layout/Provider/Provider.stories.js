import React from "react";
import { storiesOf } from "@storybook/react";
import Provider from "./Provider";

storiesOf("components/Provider", module).addWithJSX("default Provider", () => (
  <Provider>
    <h1>Default </h1>
    <p>
      Lorem ipsum
      <span>
        <a href="#"> dolor </a>
      </span>{" "}
      sip amet
    </p>
  </Provider>
));
