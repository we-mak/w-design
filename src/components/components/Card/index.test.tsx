import * as React from "react";
import { cleanup, render } from "@testing-library/react";
import "jest-styled-components";
import Card, { CardMedia, CardHeader, CardContent, CardFooter } from "./index";

import Provider from "../../layout/Provider";

afterEach(cleanup);

describe("<Button/>", () => {
  it("should render with different appearance style", () => {
    const { container } = render(
      <Provider>
        <Card raised>
          <CardHeader title="Test title" description="Lorem ipsum">
            Header
          </CardHeader>

          <CardMedia
            imageUrl="https://dak95nwic4sny.cloudfront.net/73/cheetah-safari-41193237-1554105425-ImageGalleryLightboxLarge.jpg"
            alt="test"
          />
          <CardContent>
            A safari is an overland journey, usually a trip by tourists in Africa. In the past, the
            trip was often a big-game hunt, but today, safaris are often to observe
          </CardContent>

          <CardFooter>Footer</CardFooter>
        </Card>

        <Card isLoading />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
