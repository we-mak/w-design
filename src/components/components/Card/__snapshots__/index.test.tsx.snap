// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should render with different appearance style 1`] = `
.c2 {
  font-weight: 500;
  font-family: Helvetica,Arial,sans-serif;
  font-size: 0.9rem !important;
}

.c1 {
  padding: 0.8rem;
  position: relative;
}

.c1:not(:first-child) {
  padding-bottom: 0;
}

.c1 h3 {
  font-weight: 500;
  margin-bottom: 0.2em;
}

.c3 {
  color: #566E82;
  font-size: 0.7rem;
}

.c4:first-child {
  padding-top: 0;
}

.c4:first-child img {
  border-top-left-radius: 0.2rem;
  border-top-right-radius: 0.2rem;
}

.c5 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  padding: 0.8rem;
}

.c5:not(:last-child) {
  padding-bottom: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: #fff;
  border-radius: 0.2rem;
  box-shadow: 0 2px 1px -1px rgba(0,0,0,.2),0 1px 1px 0 rgba(0,0,0,.14),0 1px 3px 0 rgba(0,0,0,.12);
  -webkit-transition: box-shadow 0.2s cubic-bezier(0.5,0.045,0.4,1);
  transition: box-shadow 0.2s cubic-bezier(0.5,0.045,0.4,1);
}

.c0:hover {
  box-shadow: 0 5px 5px -3px rgba(0,0,0,.2),0 8px 10px 1px rgba(0,0,0,.14),0 3px 14px 2px rgba(0,0,0,.12);
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: #fff;
  border-radius: 0.2rem;
  box-shadow: 0 2px 1px -1px rgba(0,0,0,.2),0 1px 1px 0 rgba(0,0,0,.14),0 1px 3px 0 rgba(0,0,0,.12);
}

.c8 {
  font-size: 0;
  line-height: 0;
  width: 100%;
}

.c9 {
  background: #BECEDC;
  width: 100%;
  min-height: 180px;
  -webkit-animation: iazffo 1.5s infinite;
  animation: iazffo 1.5s infinite;
}

.c10 {
  padding: 0.8rem;
  display: block;
  width: 100%;
  opacity: 0.1;
  -webkit-animation: iazffo 1.5s infinite;
  animation: iazffo 1.5s infinite;
}

.c10 h3 {
  background: #6C869B;
  background-size: 400% 100%;
  width: 50%;
  height: 1rem;
  border-radius: 0.5rem;
}

.c10 ul {
  width: 100%;
  list-style: none;
  padding: 0;
}

.c10 li {
  background: #BECEDC;
  background-size: 400% 100%;
  height: 1rem;
  margin: 0.4rem 0;
  border-radius: 0.5rem;
}

.c6 {
  padding: 0.8rem;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h3
        class="c2"
      >
        Test title
      </h3>
      <div
        class="c3"
      >
        Lorem ipsum
      </div>
      Header
    </div>
    <div
      class="c4"
    >
      <picture>
        <source
          data-srcset="https://photo2.foodgawker.com/wp-content/uploads/2019/08/3480441.jpg"
          media="(max-width: 480px)"
          srcset="data:image/gif;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mP88R8AAvUB+VkkrXoAAAAASUVORK5CYII="
        />
        <source
          data-srcset="https://photo2.foodgawker.com/wp-content/uploads/2019/08/3480425.jpg"
          media="(max-width: 680px)"
          srcset="data:image/gif;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mP88R8AAvUB+VkkrXoAAAAASUVORK5CYII="
        />
        <img
          alt=""
          data-src="https://photo.foodgawker.com/wp-content/uploads/2019/04/3425217.jpg"
          src="data:image/gif;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mP88R8AAvUB+VkkrXoAAAAASUVORK5CYII="
        />
      </picture>
    </div>
    <div
      class="c5"
    >
      A safari is an overland journey, usually a trip by tourists in Africa. In the past, the trip was often a big-game hunt, but today, safaris are often to observe
    </div>
    <div
      class="c6"
    >
      Footer
    </div>
  </div>
  <div
    class="c7"
  >
    <div
      class="c8"
    >
      <div
        class="c9"
      />
      <div
        class="c10"
      >
        <h3 />
        <ul>
          <li />
          <li />
        </ul>
      </div>
    </div>
  </div>
</div>
`;
